iib/workers/tasks/opm_operations.py

```diff
@@ -918,3 +930,258 @@ def verify_operator_exists(
+    bundles: List[str],
+    base_dir: str,
+    binary_image: str,
+    from_index: str,
+    overwrite_target_index_token: Optional[str] = None,
+    container_tool: Optional[str] = None,
+) -> None:
+    """
+    Deprecate the specified bundles from the index image.
+
+    Only Dockerfile is created, no build is performed.
+
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str base_dir: base directory where operation files will be located.
+    :param str binary_image: binary image to be used by the new index image.
+    :param str from_index: index image, from which the bundles will be deprecated.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    from iib.workers.tasks.utils import run_cmd, set_registry_token
+
+    cmd = [
+        Opm.opm_version,
+        'index',
+        'deprecatetruncate',
+        '--generate',
+        '--binary-image',
+        binary_image,
+        '--from-index',
+        from_index,
+        '--bundles',
+        ','.join(bundles),
+        '--allow-package-removal',
+    ]
+    if container_tool:
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+    with set_registry_token(overwrite_target_index_token, from_index):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to deprecate the bundles')
+
+
+def _find_index_version(index_image: str) -> str:
+    """
+    Find the OCP version of the given index image.
+
+    :param str index_image: index_image pull spec
+    :return: version of index image
+    :rtype: str
+
+    """
+    from iib.workers.tasks.utils import get_image_label
+
+    index_image_version = index_image.split(":")[1]
+    if index_image_version.startswith("v4."):
+        return index_image_version
+    else:
+        index_version = get_image_label(index_image, 'com.redhat.index.delivery.version')
+        return index_version
+
+
+class Opm:
+    """A class to store the opm version for the IIB operation."""
+
+    opm_version = get_worker_config().get('iib_default_opm')
+
+    @classmethod
+    def set_opm_version(cls, from_index: Optional[str] = None):
+        """
+        Set the opm version to be used for the entire IIB operation.
+
+        opm version is based on from_index/target_index.
+
+        :param str from_index: from_index_image for the request
+        """
+        log.info("Determining the OPM version to use")
+        opm_versions_config = get_worker_config().get('iib_ocp_opm_mapping')
+        if opm_versions_config is None or from_index is None:
+            log.warning(
+                "Either iib_ocp_opm_mapping config or from_index is not set, using the default opm"
+            )
+            return
```

Comment:
With this I understand we should enforce the definition of the `iib_default_opm`, no?

Because the `Opm.opm_version` uses a `get()` operation to retrieve the value from config which might be `None`, so we must be certain that it isn't.

Reply:
Also I believe it would be nice to print the default `opm` version at least in this warning.

Reply:
The default opm version would be printed in common_utils.py get_binary_versions()

Reply:
updated the config for validating the default setting as well